### PR TITLE
219: Feed Page

### PR DESCRIPTION
**Pull request by MiteshiOS**
## Task

- [219: Feed Page](https://trello.com/c/nsp3i2HL/219-feed-page)

## Description & Notes

- Added UI of "Sort By" in FeedViewController & SeeAllViewController
- Added UI of Grid View & List View in FeedViewController & SeeAllViewController
- Developed Single Product View (Added VariantCell)
- Completed Sort By Implementation in FeedViewController (Pending in SeeAllViewController due to API)

## Visuals

https://user-images.githubusercontent.com/17427703/178658529-a3cae62a-8f0f-4bbf-9e69-888578add4f8.mp4

### CODE REVIEW

**Review by tejen — CHANGES_REQUESTED** [review]
A few changes requested, and then you can re-request review.

File: archiv333/Common/ArchivOptionPopUp/ArchivOptionListViewController.swift [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return Filter.Sorting.allCases.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "OptionItemCell", for: indexPath) as! OptionItemCell
+        cell.sorting = Filter.Sorting.allCases[indexPath.row]
+        cell.ivAccessory.image = selectedSortOption == Filter.Sorting.allCases[indexPath.row] ? UIImage(systemName: "checkmark") : nil
+        cell.selectionStyle = .none
```
[the remark this answers is not in the document]
Comment (MiteshiOS) [reply archiv333/Common/ArchivOptionPopUp/ArchivOptionListViewController.swift:82]: Yes, that will also be fine. Actually, we are showing check mark already for the selected item, that's why I won't use style.

[the remark this answers is not in the document]
Comment (MiteshiOS) [reply archiv333/Common/ArchivOptionPopUp/ArchivOptionListViewController.swift]: @tejen  https://github.com/archiv333/archiv333-ios/pull/55/commits/e23cf9485a42de62dd61aa1ffb935f0f7ed75c4f

File: archiv333/Views/Search Screen Cells/SeeAllViewController.swift [hunk 2]
```diff
@@ -69,37 +85,113 @@ class SeeAllViewController: UIViewController {
...
+        vc.delegate = self
+        guard let popoverPresentationController = vc.popoverPresentationController else { fatalError("Set Modal presentation style") }
+        popoverPresentationController.sourceView = source
+        popoverPresentationController.delegate = self
+        self.present(vc, animated: true, completion: nil)
+    }
+
+    func fetchData() {
+   
+    }
```
[the remark this answers is not in the document]
Comment (MiteshiOS) [reply archiv333/Views/Search Screen Cells/SeeAllViewController.swift]: @tejen https://github.com/archiv333/archiv333-ios/pull/55/commits/e23cf9485a42de62dd61aa1ffb935f0f7ed75c4f

File: archiv333/Views/Search Screen Cells/SeeAllViewController.swift [hunk 3]
```diff
@@ -69,37 +85,113 @@ class SeeAllViewController: UIViewController {
...
+                cell.leftProduct = products[leftItemIndex]
+                cell.rightProduct = products.count - 1 > leftItemIndex ? products[leftItemIndex + 1] : nil
+            case .variants:
+                cell.leftVariant = variants[leftItemIndex]
+                cell.rightVariant = variants.count - 1 > leftItemIndex ? variants[leftItemIndex + 1] : nil
+            }
+
+            return cell
+
+        }else{
```
[the remark this answers is not in the document]
Comment (MiteshiOS) [reply archiv333/Views/Search Screen Cells/SeeAllViewController.swift]: @tejen https://github.com/archiv333/archiv333-ios/pull/55/commits/b00fe933c51c256038ad1ab705f43750975ccbe7

File: archiv333/View Controllers/Feed/FeedViewController.swift [hunk 4]
```diff
@@ -95,15 +149,36 @@ extension FeedViewController: UITableViewDelegate, UITableViewDataSource {
...
-        cell.rightProduct = products.count - 1 > leftItemIndex ? products[leftItemIndex + 1] : nil
-        return cell
+        if btnGrid.isSelected {
+            let leftItemIndex = 2 * indexPath.row
+            let cell = tableView.dequeueReusableCell(withIdentifier: "VariantsCell", for: indexPath) as! VariantsCell
+            cell.delegate = self
+            cell.leftProduct = products[leftItemIndex]
+            cell.rightProduct = products.count - 1 > leftItemIndex ? products[leftItemIndex + 1] : nil
+            return cell
+        }else{
```
[the remark this answers is not in the document]
Comment (MiteshiOS) [reply archiv333/View Controllers/Feed/FeedViewController.swift]: @tejen https://github.com/archiv333/archiv333-ios/pull/55/commits/bb160e484b14016ad94d0334b7e40d91c268fc55

**Review by tejen — APPROVED** [review]

**Pull request merged**